src/lava/magma/compiler/builder.py

```diff
@@ -326,13 +353,53 @@ def build(self):
                 csp_ports = self.csp_ports[name]
                 if not isinstance(csp_ports, list):
                     csp_ports = [csp_ports]
-            port = port_cls(pm, csp_ports, p.shape, lt.d_type)
+            port = port_cls(csp_ports, pm, p.shape, lt.d_type)
 
             # Create dynamic PyPort attribute on ProcModel
             setattr(pm, name, port)
             # Create private attribute for port precision
             # setattr(pm, "_" + name + "_p", lt.precision)
 
+        # Initialize RefPorts
+        for name, p in self.ref_ports.items():
+            # Build PyPort
+            lt = self._get_lava_type(name)
+            port_cls = ty.cast(ty.Type[PyRefPort], lt.cls)
+            csp_recv = None
+            csp_send = None
+            if name in self.csp_ports:
+                csp_ports = self.csp_ports[name]
+                csp_recv = csp_ports[0] if isinstance(
```

Comment:
This is not so elegant and looks like a hack. Normally there could be multiple csp_ports in case of 1:many or many:1 connectivity. 
Here there are multiple ports because we've just misused it to pass the different send and recv ports. That's conceptually different. If we throw errors that forbid non 1:1 connections then this will work but we should find a better solution. 

Either implement a better way of file a descriptive issue to address this hack and make it cleaner.

Reply:
Issue #59 filed.